### PR TITLE
Version upgrade to es 8.11.4

### PR DESCRIPTION
**Pull request by Xiaomeng-Faire**
Credit to Guilherme for upgrading it to 8.9.2 in his [pr](https://github.com/o19s/elasticsearch-learning-to-rank/pull/473).

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ltrVersion = 1.5.8
-elasticsearchVersion = 8.9.2
+elasticsearchVersion = 8.11.1
 luceneVersion = 9.7.0
```
Comment (wrigleyDan) [inline gradle.properties:3]: Elasticsearch version 8.11.1 uses Lucene version 9.8.0.
Please change to `luceneVersion = 9.8.0`

File: gradle.properties [hunk 2]
```diff
@@ -1,5 +1,5 @@
 ltrVersion = 1.5.8
-elasticsearchVersion = 8.9.2
-luceneVersion = 9.7.0
+elasticsearchVersion = 8.11.1
```
Comment (Philippus) [inline gradle.properties:2]: Latest version is 8.11.2 I think we can skip 8.11.1.

**Comment by wrigleyDan** [comment]
There have been quite some changes in Elasticsearch that now conflict with the plugin code. One major commit being the removal of StatusToXContent and RestStatusToXContentListener that are both used: https://github.com/elastic/elasticsearch/commit/f580501bc98cbc78d67c2223f8020a271a631c8f

**Comment by wrigleyDan** [comment]
@Xiaomeng-Faire I opened a PR for some changes that make the code compile and run through the tests successfully. As soon as you merge them into your PR we can get this moving again.
Thanks!

**Pull request merged**